src/network/networkd-link.c

```diff
@@ -87,9 +87,12 @@ bool link_ipv4ll_enabled(Link *link) {
         if (ether_addr_is_null(&link->hw_addr.ether))
                 return false;
 
-        if (STRPTR_IN_SET(link->kind,
-                          "vrf", "wireguard", "ipip", "gre", "ip6gre","ip6tnl", "sit", "vti",
-                          "vti6", "nlmon", "xfrm", "bareudp"))
```

Comment:
So it works on e.g. "bareudp" ?

Reply:
Right. Bareudp has ARPHRD_NONE. Wireguard is ARPHRD_WIREGUARD, Tunnel devices has e.g. ARPHRD_GRE.

Reply:
Can you add some explanation to the commit message?

Reply:
Done. PTAL.